andela/vidar-ah-backend
# #164139683 Add password reset feature

#### What does this PR do?
This feature adds a feature for a user to reset password
#### Description of task to be completed?
* Add `passwordResetToken` and `passwordResetTokenExpires` columns to `User` table
* Add endpoint to send reset email
* Add endpoint to verify password reset link
* Add endpoint to change user password
* Write tests
* Add authentication class
#### How should this be manually tested?
* Run `npm run db-migrate` to run migrations
* Run `npm start` to start the application
* Create a user by sending a post request to `/api/v1/user`
* Verify the user by clicking the link in your email
* Send a post request with your email to `/api/v1/reset` with your email as post body `email: 'email@domain.com'`
* Click the link sent to your email
* Copy the token in the request response
* Send token along with your new password to `api/v1/reset/3`
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/164139683
